packages/@react-aria/select/src/useSelect.ts

```diff
@@ -161,6 +162,7 @@ export function useSelect<T>(props: AriaSelectOptions<T>, state: SelectState<T>,
         }
 
         if (props.onBlur) {
+          props.onFocusChange(true);
```

Comment:
This should be
```suggestion
          props.onFocusChange(false);
```
since we are blurring away from the select element. Same for the `onFocusChange` down below for the menu